test/jdk/javax/swing/JTextArea/bug4849868.java

```diff
@@ -0,0 +1,105 @@
+/*
+ * Copyright (c) 2003, 2023, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+/* @test
+ * @bug 4849868
+ * @summary Tests if JTextArea.getSelectionEnd works correctly
+ * @key headful
+ * @run main bug4849868
+ */
+
+import java.awt.AWTException;
+import java.awt.Point;
+import java.awt.Robot;
+import java.awt.event.InputEvent;
+import java.lang.reflect.InvocationTargetException;
+
+import javax.swing.JFrame;
+import javax.swing.JScrollPane;
+import javax.swing.JTextArea;
+import javax.swing.SwingUtilities;
+
+public class bug4849868 {
+
+    private static volatile boolean passed = false;
+
+    private static JTextArea textArea;
+    private static JFrame f;
+    private static Point p;
+
+    private static int end;
+    private static int len;
+
+    public static void main(String[] args) throws InterruptedException,
+            InvocationTargetException, AWTException {
+        try {
+            Robot robot = new Robot();
+            robot.setAutoDelay(100);
+            robot.setAutoWaitForIdle(true);
+
+            SwingUtilities.invokeAndWait(() -> {
+                f = new JFrame("bug4849868");
+                textArea = new JTextArea("1234");
+                textArea.setLineWrap(true);
+                JScrollPane pane = new JScrollPane(textArea,
+                        JScrollPane.VERTICAL_SCROLLBAR_ALWAYS,
+                        JScrollPane.HORIZONTAL_SCROLLBAR_NEVER);
+                f.getContentPane().add(pane);
+                f.setSize(300, 300);
+                f.setVisible(true);
+                f.setLocationRelativeTo(null);
+            });
+
+            robot.delay(1000);
+
+            SwingUtilities.invokeAndWait(() -> p =
+                    textArea.getLocationOnScreen());
```

Comment:
```suggestion
            SwingUtilities.invokeAndWait(() -> 
                    p = textArea.getLocationOnScreen());
```
I am absolutely sure it's better not to break the assignment in lambda expressions, especially when the variable name is a single letter.